[v6] deprecate import.meta.env.ASSETS_PREFIX (#12468)
withastro/docs: src/content/docs/en/guides/upgrade-to/v6.mdx

Need to upgrade an older project to v5 first? See our [older migration guide](/en/guides/upgrade-to/v5/).

> Need to see the v5 docs? Visit this [older version of the docs site (unmaintained v5.0 snapshot)](https://v5.docs.astro.build/).

Reviewer: Yes, Armand is correct. We're not sure what the latest version of 5 will be yet, and why we usually start all the breaking changes with "In Astro 5.x..." meaning, "all the fives". (You can see the last upgrade guide to see that the published version does in fact say "In Astro 4.x" everywhere!

But *this one* we will update to say specifically what the last version of Astro 5 was, once it's known.

```suggestion
Need to see the v5 docs? Visit this [older version of the docs site (unmaintained v5.xx snapshot)](https://v5.docs.astro.build/).
```

Some deprecated features may temporarily continue to function until they are completely removed. Others may silently have no effect, or throw an error prompting you to update your code.

### Deprecated: `Astro` in `getStaticPaths()`

<SourcePR number="14432" title="feat: deprecate Astro in getStaticPaths"/>

> In Astro 5.0, it was possible to access an `Astro` object inside `getStaticPaths()`. However, despite being typed the same as the `Astro` object accessible in the frontmatter, this object only had `site` and `generator` properties. This could lead to confusion about which `Astro` object properties were available inside `getStaticPaths()`.

Reviewer: ```suggestion
In Astro 5.x, it was possible to access an `Astro` object inside `getStaticPaths()`. However, despite being typed the same as the `Astro` object accessible in the frontmatter, this object only had `site` and `generator` properties. This could lead to confusion about which `Astro` object properties were available inside `getStaticPaths()`.
```

<ReadMore>Read more about [built-in environment variables such as `import.meta.env.SITE`](/en/guides/environment-variables/#default-environment-variables) that are accessible when [using `getStaticPaths()` to dynamically generate static routes](/en/guides/routing/#static-ssg-mode).</ReadMore>

### Deprecated: `import.meta.env.ASSETS_PREFIX`

<SourcePR number="14461" title="feat: deprecate import.meta.env.ASSETS_PREFIX"/>

> In Astro 5.0, it was possible to access [`build.assetsPrefix`] via `import.meta.env.ASSETS_PREFIX`. It could be a string or an object, which is not very standard when it comes to `import.meta.env` values.

Reviewer: ```suggestion
In Astro 5.x, it was possible to access [`build.assetsPrefix`] in your Astro config via the built-in environment variable `import.meta.env.ASSETS_PREFIX`. However, Astro v5.7.0 introduced the `astro:config` virtual model to expose a non-exhaustive, serializable, type-safe version of the Astro configuration which included access to `build.assetsPrefix` directly. This became the preferred way to access the prefix for Astro-generated asset links when set, although the environment variable still existed.
```

Astro 6.0 deprecates this variable in favor of `build.assetsPrefix` from the `astro:config/server` module.

#### What should I do?

> Replace your usage of `import.meta.env.ASSETS_PREFIX` by `astro:config/server`:

Reviewer: ```suggestion
Replace any occurances of `import.meta.env.ASSETS_PREFIX` with the `build.assetsPrefix` import from `astro:config/server`. This is a drop-in replacement to provide the existing value, and no other changes to your code should be necessary:
```

<SourcePR number="14400" title="Remove deprecated ViewTransitions component"/>

> In Astro 5.0, the `<ViewTransitions />` component was renamed to `<ClientRouter />` to clarify the role of the component. The new name makes it more clear that the features you get from Astro's `<ClientRouter />` routing component are slightly different from the native CSS-based MPA router. However, a deprecated version of the `<ViewTransitions />` component still existed and may have functioned in Astro 5.0.

Reviewer: ```suggestion
In Astro 5.0, the `<ViewTransitions />` component was renamed to `<ClientRouter />` to clarify the role of the component. The new name makes it more clear that the features you get from Astro's `<ClientRouter />` routing component are slightly different from the native CSS-based MPA router. However, a deprecated version of the `<ViewTransitions />` component still existed and may have functioned in Astro 5.x.
```
